apps/aechannel/src/aesc_fsm.erl

```diff
@@ -2595,10 +2716,25 @@ has_my_signature(Me, SignedTx) ->
             ok =:= aetx_sign:verify_one_pubkey(Me, SignedTx, CurrHeight)
     end.
 
-start_min_depth_watcher(Type, SignedTx, Updates,
+
+maybe_restart_watcher(#data{ watcher = undefined
```

Comment:
I like the assertive approach but possibly drop the `maybe_` prefix as it is a bit misleading: we either start a new watcher or we crash.